app/src/main/java/com/google/android/soundchecker/utils/AudioTrackSink.kt

```diff
@@ -66,7 +72,9 @@ class AudioTrackSink(var preferredDevice: AudioDeviceInfo? = null,
     }
 
     override fun start() {
-        open()
+        if (open() != AudioErrorCallback.SUCCESS) {
+            return
```

Comment:
Are we failing silently?

Reply:
When it fails, the caller knows via error callback.